### PR TITLE
EPL-7782: Allow Async Fetch of Refs Within Keywords + Resolve Fragments

### PR DESCRIPTION
**Pull request by michaelcarter-wf**
## Ultimate problem:
- The library doesn't resolve schema correctly when they have remote (read: http) `$ref`s nested within `properties` and other sub schemas, like `items` or `oneOf` / `anyOf` / `allOf`.
- When keyword refs also specify path fragments, they aren't resolved correctly.

Note: We don't yet handle refs within refs, further down the 🐇 🕳  we go in PR 3, stay tuned!
 
## How it was fixed:
Blood, sweat, tears, and lots of recursion. 🤣 

## Testing suggestions:
- Pull down locally and run: `ddev format --check; ddev test`

## Potential areas of regression:

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Module Findings
No security relevant content detected. Please review for security relevance and request security review as needed.

**Comment by semveraudit-wf** [comment]
### Public API Changes

 No changes to the public API found for commit f11511dc2e829c3a91a1356fa57b1940f51688aa

Showing results for f11511dc2e829c3a91a1356fa57b1940f51688aa
> Powered by [semver-audit-service](https://github.com/Workiva/semver-audit-service). Please report any problems by [filing an issue](https://github.com/Workiva/semver-audit-service/issues).
[Browse public API](https://semver-audit.wk-dev.wdesk.org/Workiva/json_schema/f11511dc2e829c3a91a1356fa57b1940f51688aa).

**Comment by jamesromo-wf** [comment]
All the tests pass. The formatter returns:
```
The Dart formatter needs to be run. The following files require changes:
    test/unit/vm/json_schema/validation_test.dart
```

File: lib/src/json_schema/json_schema.dart [hunk 1]
```diff
@@ -184,55 +185,136 @@ class JsonSchema {
...
+  /// The path of the [JsonSchema] within the root [JsonSchema].
   String _path;
+
+  /// Title of the [JsonSchema].
   String _title;
+
+  /// List of allowable types for the [JsonSchema].
   List<SchemaType> _schemaTypeList;
 
+  // --------------------------------------------------------------------------
```
Comment (jamesromo-wf) [inline lib/src/json_schema/json_schema.dart]: Is this how we want to do Code "Sections"? Been struggling to find dart consistency with these.

File: lib/src/json_schema/vm/platform_functions.dart [hunk 2]
```diff
@@ -41,24 +41,27 @@ import 'dart:async';
...
-        final map = convert.JSON.decode(schemaText);
-        return JsonSchema.createSchema(map);
-      });
-    });
+    // Setup the HTTP request.
+    final httpRequest = await new HttpClient().getUrl(uri);
+    httpRequest.followRedirects = true;
+    // Fetch the response
+    final response = await httpRequest.close();
+    // Convert the response into a string;
```
Comment (jamesromo-wf) [inline lib/src/json_schema/vm/platform_functions.dart]: ```;``` intentional here?

File: test/unit/vm/json_schema/validation_test.dart [hunk 3]
```diff
@@ -94,6 +103,7 @@ void main([List<String> args]) {
               final bool expectedResult = validationTest['valid'];
               final checkResult = expectAsync0(() => expect(validationResult, expectedResult));
               JsonSchema.createSchema(schemaData).then((schema) {
+                print(schema.refMap);
```
Comment (jamesromo-wf) [inline test/unit/vm/json_schema/validation_test.dart]: Should this be a log instead?

File: lib/src/json_schema/json_schema.dart (hunk 1 above)
Comment (michaelcarter-wf) [reply lib/src/json_schema/json_schema.dart]: I use the one's from Jace's over react snippets package. No opinion here, I'd be happy to settle on these.

**Pull request merged**